pulser-core/pulser/channels/comparison_tools.py

```diff
@@ -0,0 +1,201 @@
+        rightvalues: Dict of values on the right of the comparison operator
+
+    Returns:
+        Dictionary having the keys of the comparison operator, associating
+        None if both values are None for this key, and the result of the
+        comparison otherwise.
+    """
+    # Error if some keys of comparison operators are not dict of left or right
+    if not (
+        comparison_ops.keys() <= leftvalues.keys()
+        and comparison_ops.keys() <= rightvalues.keys()
+    ):
+        raise ValueError(
+            "Keys in comparison_ops should be in left values and right values."
+        )
+    # Compare using +inf and -inf to replace None values
+    return {
+        key: comparison_op(
+            *(
+                value
+                or (
+                    float("+inf")
+                    if comparison_op in [lt, min]
+                    else float("-inf")
+                )
+                for value in (leftvalues[key], rightvalues[key])
+            )
+        )
+        if (leftvalues[key], rightvalues[key]) != (None, None)
+        else None
+        for (key, comparison_op) in comparison_ops.items()
+    }
+
+
+def _validate_obj_from_best(
+    obj_dict: dict, best_obj_dict: dict, comparison_ops: dict
+) -> bool:
+    """Validates an object by comparing it with a better one.
+
+    Attributes:
+        obj_dict: Dict of attributes and values of the object to compare.
+        best_obj_dict: Dict of attributes and values of the best object.
+        comparison_ops: Dict of attributes and comparison operators to
+            use to compare the object and the best object.
+
+    Returns:
+        True if the comparison works, raises a ValueError otherwise.
+    """
+    # If the two values are almost equal then there is no need to compare them
+    comparison_ops_keys = list(comparison_ops.keys())
+    for key in comparison_ops_keys:
+        if (
+            obj_dict[key] is not None
+            and best_obj_dict[key] is not None
+            and isinstance(obj_dict[key], numbers.Number)
+            and isinstance(best_obj_dict[key], numbers.Number)
+            and np.isclose(obj_dict[key], best_obj_dict[key], rtol=1e-14)
+        ):
+            comparison_ops.pop(key)
+    is_wrong_effective_ch = _compare_with_None(
+        comparison_ops, obj_dict, best_obj_dict
+    )
+    # Validates if no True in the dictionary of the comparisons
+    if not (True in is_wrong_effective_ch.values()):
+        return True
+    is_wrong_effective_index = list(is_wrong_effective_ch.values()).index(True)
+    is_wrong_key = list(is_wrong_effective_ch.keys())[is_wrong_effective_index]
+    raise ValueError(
+        f"{is_wrong_key} cannot be"
+        + (
+            " below "
+            if comparison_ops[is_wrong_key] == lt
+            else (
+                " above "
+                if comparison_ops[is_wrong_key] == gt
+                else " different than "
+            )
+        )
```

Comment:
nit: this would probably be more readable if it were defined in a variable beforehand